pyspectral/tests/test_reflectance.py

```diff
@@ -221,3 +215,33 @@ def test_reflectance(self):
             self.assertTrue(hasattr(refl, 'compute'))
         except ImportError:
             pass
+
+
+def test_get_as_array_from_scalar_input():
+    """Test the function to return an array when input is a scalar"""
+    res = get_as_array(1)
+
+    if hasattr(res, 'compute'):
+        assert res.compute()[0] == 1
+    else:
+        assert res[0] == 1
```

Comment:
Should this Numpy case be tested in every case? It should be possible by patching, something like this:
```python
with mock.patch('from pyspectral.near_infrared_reflectance.array', new=np.ndarray):
    res = get_as_array(1)
    assert res[0] == 1
```
It might be necessary to move the imports inside the testing functions. Which is better practice anyway when patching is needed.

Reply:
As said below @pnuu I don't see how to patch the numpy `array` function if this doesn't work:
```
with patch('pyspectral.near_infrared_reflectance.array', new=np.array):
    res = get_as_array(1)
    assert res[0] == 1
```
And it doesn't behave as I expected. The code is using the built-in `array` not the Numpy one!
Any ideas?

Reply:
Ok. So I was fooled. By this debugging:
```
ipdb> array
<built-in function array>
ipdb> array.__module__
'numpy'
```
So, the array patching above that you suggested @pnuu works, is my conclusion.
But now I actually changed the function to always use `asanyarray`
Do you see any problem is that @djhoese or @pnuu ?

Reply:
No, but I'm not really sure everything that gets done with this so it is worth a shot.

Reply:
Ok, thanks!